quiche/src/dgram.rs

```diff
@@ -89,11 +107,11 @@ impl DatagramQueue {
     }
 
     pub fn has_pending(&self) -> bool {
-        !self.queue.as_ref().map(|q| q.is_empty()).unwrap_or(true)
+        !self.queue.iter().next().map(|(_, q)| q.is_empty()).unwrap_or(true)
     }
 
     pub fn purge<F: Fn(&[u8]) -> bool>(&mut self, f: F) {
-        if let Some(q) = self.queue.as_mut() {
+        for (_, q) in self.queue.iter_mut() {
```

Comment:
now that we're looping, `self.queue_bytes_size` will end up reflecting the _last_ queue size, not the total. Maybe set to zero before the loop and accumulate each queue size as you loop?